libs/net/core/Http/HttpRequestClient.cs

```diff
@@ -78,20 +79,29 @@ public void Dispose()
             var responseStream = await response.Content.ReadAsStreamAsync();
             // var data = await response.Content.ReadAsByteArrayAsync();
             var contentType = response.Content.Headers.ContentType;
+            var body = "";
             try
             {
-                if (contentType?.MediaType?.Contains("json", StringComparison.InvariantCultureIgnoreCase) == true)
+                if (contentType?.MediaType?.Contains("json", StringComparison.InvariantCultureIgnoreCase) == true && response.IsSuccessStatusCode)
                     return JsonSerializer.Deserialize<TModel>(responseStream, _serializeOptions);
+                else if (response.IsSuccessStatusCode)
+                    return default;
+                else if (response.StatusCode == HttpStatusCode.NotModified)
```

Comment:
304 is not considered successful, but it actually is successful.  We must return null.